build.rs

```diff
@@ -1,4 +1,70 @@
+use std::env;
+use std::path::Path;
+use std::path::PathBuf;
+use std::process::Command;
+
+#[cfg(feature = "bundled")]
+fn fetch_hwloc(path: &Path, version: &str) {
+    if !path.join("hwloc").join("Makefile.am").exists() {
+        let r = Command::new("git")
+            .args(&[
+                "clone",
+                "https://github.com/open-mpi/hwloc",
+                "--depth",
+                "1",
+                "--branch",
+                version,
+            ])
+            .current_dir(path.clone())
+            .output();
+        let m = r.expect("Failed to run git clone");
+        assert!(
+            m.status.success(),
+            "git clone for hwloc returned failure code {}\n{:?}",
+            m.status,
+            m
+        );
+    }
+}
+
+#[cfg(feature = "bundled")]
+fn get_os_from_triple(triple: &str) -> Option<&str> {
+    triple.splitn(3, "-").nth(2)
+}
+
+#[cfg(feature = "bundled")]
+fn compile_hwloc_autotools(p: PathBuf) -> PathBuf {
+    let mut config = autotools::Config::new(p);
+    config.reconf("-ivf").make_target("install").build()
+}
+
+#[cfg(feature = "bundled")]
+fn compile_hwloc_cmake(build_path: &Path, target_os: &str) -> PathBuf {
+    let mut cfg = cmake::Config::new(build_path);
+    if let Ok(profile) = env::var("HWLOC_BUILD_PROFILE") {
+        cfg.profile(&profile);
+    } else {
+        cfg.profile("release");
+    }
+
+    // Allow specifying custom toolchain specifically for hwloc.
+    if let Ok(toolchain) = env::var("HWLOC_TOOLCHAIN") {
+        cfg.define("CMAKE_TOOLCHAIN_FILE", &toolchain);
+    }
+    cfg.build()
+}
+
+fn use_pkgconfig(required_version: &str) -> pkg_config::Library {
+    let lib = pkg_config::Config::new()
+        .range_version(required_version.."3.0.0")
+        .statik(true)
+        .probe("hwloc")
+        .expect("Could not find a suitable version of hwloc");
+    lib
+}
+
 fn main() {
+    let git_dir = env::var("OUT_DIR").expect("No output directory given");
```

Comment:
Should move into the `#[cfg(feature = "bundled")]` section to avoid an unused variable warning in the non-bundled case;

Reply:
Right, get rid of warnings for both cases.